### PR TITLE
Modify rule S2631: Change text to the education framework format [Python][APPSEC-288]

### PR DESCRIPTION
**Pull request by pedro-oliveira-sonarsource**

### CODE REVIEW

File: rules/S2631/common/fix/validation.adoc [hunk 1]
```diff
@@ -1,16 +1,17 @@
-==== Sanitization and validation
+==== Sanitization and Validation
 
-The easiest way to protect against regex injections is to escape metacharacters,
- preferably with native functions. +
-The escape function instructs the regular expression engine to interpret these
-characters literally and not as metacharacters.
+Metacharacters escape using native functions is a solution against regex
+injection. + The escape function sanitizes the input so that the regular expression
```
Comment (egon-okerman-sonarsource) [inline rules/S2631/common/fix/validation.adoc:4]: The `+`'es in this document don't act how they're supposed to (they can be seen in the rendered document.) I think that to function correctly as a line break, you have to put the next sentence on a new line.

File: rules/S2631/python/rule.adoc [hunk 2]
```diff
@@ -0,0 +1,31 @@
+== Why is this an issue?
+
+include::../rationale.adoc[]
+
+include::../impact.adoc[]
```
Comment (egon-okerman-sonarsource) [inline rules/S2631/python/rule.adoc:5]: This wasn't changed in this PR, but rationale and impact have some terms (e.g. backtracking and "catastrophic backtracking situations") surrounded in code tags. I think [quotation marks should be used instead](https://github.com/SonarSource/rspec/blob/master/docs/styling_guide.adoc#literals).

File: rules/S2631/python/how-to-fix-it/python.adoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+[source,python,diff-id=1,diff-type=compliant]
+----
+from flask import request
+import re
+
+@app.route('/lookup')
+def lookup():
+  regex = request.args.get('regex')
+  data = request.get('data')
```
Comment (egon-okerman-sonarsource) [inline rules/S2631/python/how-to-fix-it/python.adoc:32]: Not sure if this is intentional, but this line changed compared to the noncompliant code sample.

**Review by egon-okerman-sonarsource — COMMENTED** [review]
Some small fixes are necessary, after that should be good to go.

File: rules/S2631/python/rule.adoc (hunk 2 above)
Comment (pedro-oliveira-sonarsource) [reply rules/S2631/python/rule.adoc:5]: I have removed them as I am not aware of the quotation mark alternative, can you apply the change to one of them to have a look at?

**Review by egon-okerman-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**